fpdf/text_region.py

```diff
@@ -0,0 +1,275 @@
+            if hasattr(self, "accept_page_break"):
+                if self.pdf.y + text_line.height > self.pdf.page_break_trigger:
+                    page_break_triggered = self.accept_page_break()
+            new_page = self.pdf._render_styled_text_line(
+                text_line,
+                text_line.max_width,
+                h=text_line.height,
+                border=0,
+                new_x=XPos.WCONT,
+                new_y=YPos.TOP,
+                fill=False,
+                #                link=link,
+            )
+            page_break_triggered = page_break_triggered or new_page
+        if text_line and text_line.trailing_nl:
+            # The line renderer can't handle trailing newlines in the text.
+            self.pdf._ln()
+        return page_break_triggered
+
+    def collect_lines(self, print_sh: bool = False):
+        text_lines = []
+        current_y = self.pdf.y
+        for paragraph in self._paragraphs:
+            cur_lines = paragraph._build_lines(current_y, print_sh)
+            if not cur_lines:
+                continue
+            current_y = paragraph.current_y
+            text_lines.extend(cur_lines)
+        return text_lines
+
+    def render(self, print_sh: bool = False):
+        if not self._paragraphs:
+            return False
+        text_lines = self.collect_lines(print_sh)
+        return self._render_lines(text_lines)
+
+    def get_width(self, height):
+        limits = self.current_x_extents(self.pdf.y, height)
+        res = limits[1] - max(self.pdf.x, limits[0]) - 2 * self.pdf.c_margin
+        return res
+
+
+class TextColumnarMixin:
+    """Enable a TextRegion to perform page breaks"""
+
+    def __init__(self, pdf, *args, l_margin=None, r_margin=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.l_margin = pdf.l_margin if l_margin is None else l_margin
+        left = self.l_margin
+        self.r_margin = pdf.r_margin if r_margin is None else r_margin
+        right = pdf.w - self.r_margin
+        self._set_left_right(left, right)
+
+    def _set_left_right(self, left, right):
+        self.left = self.pdf.l_margin if left is None else left
+        self.right = (self.pdf.w - self.pdf.r_margin) if right is None else right
+        if self.right <= self.left:
+            raise FPDFException(
+                f"{self.__class__.__name__}(): "
+                f"Right limit ({self.right}) lower than left limit ({self.left})."
+            )
+
+    def current_x_extents(self, y, height):  # pylint: disable=unused-argument
+        """Return the horizontal extents of the current line.
+        Columnar regions simply return the boundaries of the column.
+        Regions with non-vertical boundaries need to check how the largest
+        font-height in the current line actually fits in there.
+        For that reason we include the current y and the line height.
+        """
+        return self.left, self.right
```

Comment:
Currently this method is always overriden.
Is it really meaningful to define default values there?

Reply:
I'll know that once I have more different subclasses.